tests/raqm-test.c

```diff
@@ -28,152 +28,385 @@
-  g_option_context_free (context);
-
-  text = g_strcompress (text);
-
-  if (text == NULL || (font == NULL && fonts == NULL))
-  {
-    g_print ("Text or font is missing.\n\n%s",
-             g_option_context_get_help (context, TRUE, NULL));
-    return 1;
-  }
-
-  dir = RAQM_DIRECTION_DEFAULT;
-  if (direction && strcmp(direction, "rtl") == 0)
-    dir = RAQM_DIRECTION_RTL;
-  else if (direction && strcmp(direction, "ltr") == 0)
-    dir = RAQM_DIRECTION_LTR;
-
-  rq = raqm_create ();
-  assert (raqm_set_text_utf8 (rq, text, strlen (text)));
-  assert (raqm_set_par_direction (rq, dir));
-  assert (!FT_Init_FreeType (&library));
-
-  if (fonts)
-  {
-    gchar **list = g_strsplit (fonts, ",", -1);
-    for (gchar **p = list; p != NULL && *p != NULL; p++)
-    {
-      gchar **sublist = g_strsplit (*p, " ", -1);
-      gchar **range;
-      int s, l;
-      assert (!FT_New_Face (library, sublist[0], 0, &face));
-      assert (!FT_Set_Char_Size (face, face->units_per_EM, 0, 0, 0));
-      range = g_strsplit (sublist[1], ":", -1);
-      s = atoi(range[0]);
-      l = atoi(range[1]);
-      assert (raqm_set_freetype_face_range(rq, face, s, l));
-
-      g_strfreev (range);
-      g_strfreev (sublist);
-    }
-    g_strfreev (list);
-  } else {
-    assert (!FT_New_Face (library, font, 0, &face));
-    assert (!FT_Set_Char_Size (face, face->units_per_EM, 0, 0, 0));
-    assert (raqm_set_freetype_face (rq, face));
-  }
-
-  if (languages)
-  {
-    gchar **list = g_strsplit (languages, ",", -1);
-    for (gchar **p = list; p != NULL && *p != NULL; p++)
-    {
-      gchar **sublist = g_strsplit (*p, ";", -1);
-      gchar **range;
-      gchar *lang;
-      int start, length;
-
-      lang = sublist[0];
-      range = g_strsplit (sublist[1], ":", -1);
-      start = atoi(range[0]);
-      length = atoi(range[1]);
-      assert (raqm_set_language(rq, lang, start, length));
-
-      g_strfreev (range);
-      g_strfreev (sublist);
-    }
-    g_strfreev (list);
-  }
-
-  if (features)
-  {
-    gchar **list = g_strsplit (features, ",", -1);
-    for (gchar **p = list; p != NULL && *p != NULL; p++)
-      assert (raqm_add_font_feature (rq, *p, -1));
-    g_strfreev (list);
-  }
-
-  assert (raqm_layout (rq));
-
-  glyphs = raqm_get_glyphs (rq, &count);
-  assert (glyphs != NULL);
-
-  if (cluster >= 0)
-  {
-    index = cluster;
-    assert (raqm_index_to_position (rq, &index, &x, &y));
-  }
-  
-  if (position)
-    assert (raqm_position_to_index (rq, position, 0, &start_index));
-
-  raqm_destroy (rq);
-  FT_Done_Face (face);
-  FT_Done_FreeType (library);
-
-  return 0;
+    const char *testname = NULL;
+    const char *direction = NULL;
+    const char *text = NULL;
+    const char *features = NULL;
+    const char *font = NULL;
+    const char *fonts = NULL;
+    const char *languages = NULL;
+    const char *tmpFont = NULL;
+    int cluster = -1;
+    int position = -1;
+
+    testname = argv[1];
+
+    if (strcmp(testname, "test1.test") == 0){
```

Comment:
Also I’d prefer to have the test changes in a separate pull request, since we want these regardless of the build system.